arches_lingo/src/arches_lingo/components/generic/reference-datatype/ReferenceDatatypeEditor.vue

```diff
@@ -0,0 +1,87 @@
+<script setup lang="ts">
+import { computed, inject, ref } from "vue";
+import Select from "primevue/select";
+import MultiSelect from "primevue/multiselect";
+
+import { systemLanguageKey } from "@/arches_lingo/constants.ts";
+import { PREF_LABEL } from "@/arches_vue_utils/constants.ts";
+
+import type { ControlledListItem } from "@/arches_lingo/types";
+import type { Language } from "@/arches_vue_utils/types.ts";
+
+const systemLanguage = inject(systemLanguageKey) as Language;
+const props = withDefaults(
+    defineProps<{
+        value?: ControlledListItem[];
+        options?: ControlledListItem[];
+        multiValue?: boolean;
+    }>(),
+    {
+        value: () => [],
+        options: () => [],
+    },
+);
+
+const emit = defineEmits(["update"]);
+const uriRef = ref(
```

Comment:
this is pretty complex, and would be more readable/maintainable if broken out into its own function.